Week6/app/src/main/java/com/oliverperez/blockbusterchallenge/app/BlockbusterApplication.kt

```diff
@@ -2,21 +2,19 @@ package com.oliverperez.blockbusterchallenge.app
 
 import android.app.Application
 import android.content.Context
-import androidx.room.Room
 import com.oliverperez.blockbusterchallenge.model.MoviesDatabase
+import kotlinx.coroutines.CoroutineScope
 
 class BlockbusterApplication : Application() {
 
     companion object {
-        lateinit var database: MoviesDatabase
         private lateinit var instance: BlockbusterApplication
         fun getAppContext(): Context = instance.applicationContext
+        fun getDatabase(scope: CoroutineScope) = MoviesDatabase.getDatabase(instance.applicationContext, scope)
```

Comment:
You should call getDatabase only once and store the reference in a variable to use it later on.